### PR TITLE
Make sitrep panel icon and text size adjustable

### PR DESCRIPTION
**Pull request by TheSilentOne1**
Makes sitrep panel icon and text size individually adjustable via
config.xml.

### CODE REVIEW

File: UI/SitRepPanel.cpp [hunk 1]
```diff
@@ -202,12 +206,16 @@ namespace {
...
             if (m_link_text) {
-                // Use the height of the text as our height.
+                // establish sitrep panel's size; use icon or text size for panel height, whichever is larger
                 // DoLayout reflowed the text.
-                GG::Pt text_size = m_link_text->TextLowerRight() - m_link_text->TextUpperLeft();
-                text_size.y += ITEM_VERTICAL_PADDING*2; // Text centers, so this puts padding on both above and below
-                text_size.x = lr.x - ul.x; // Ignore the width of the text, use whatever was requested.
-                GG::Control::SizeMove(ul, ul + text_size );
+                GG::Pt panel_text = m_link_text->TextLowerRight() - m_link_text->TextUpperLeft();
+                int text_size = Value(panel_text.y);
```
Comment (Dilvish-fo) [inline UI/SitRepPanel.cpp]: "panel_text" seems like a wrong name for something that is more of a text size, and you could simply combine these lines: 

```
int text_size = Value((m_link_text->TextLowerRight() - m_link_text->TextUpperLeft()).y);
```

Also, just to mention an alternative approach using different info that you might more specifically want to use sometime, you can also get a working result with 

```
int text_size = Value(m_link_text->GetFont()->Lineskip()) * m_link_text->GetLineData().size();
```

**Comment by TheSilentOne1** [comment]
Thanks for your feedback, how would I best go about changing that line?
Change it in my local private repo > push to my github repo > new pull request?
Or is there a more direct approach?

**Comment by Dilvish-fo** [comment]
> Or is there a more direct approach?

yes, just make the change within your private repo  (in the same branch that you used for the PR), then push that to your github repo and then github will automatically update the PR with the additional commit

**Comment by TheSilentOne1** [comment]
Thanks, there it is.

File: UI/SitRepPanel.cpp [hunk 2]
```diff
@@ -202,12 +206,15 @@ namespace {
...
-                // Use the height of the text as our height.
+                // establish sitrep panel's size; use icon or text size for panel height, whichever is larger
                 // DoLayout reflowed the text.
-                GG::Pt text_size = m_link_text->TextLowerRight() - m_link_text->TextUpperLeft();
-                text_size.y += ITEM_VERTICAL_PADDING*2; // Text centers, so this puts padding on both above and below
-                text_size.x = lr.x - ul.x; // Ignore the width of the text, use whatever was requested.
-                GG::Control::SizeMove(ul, ul + text_size );
+                int text_size = Value((m_link_text->TextLowerRight() - m_link_text->TextUpperLeft()).y);
+                int icon_size = Value(GetIconSize());
+                int maxPanelSize = std::max(text_size, icon_size);
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp]: name variables with all_lower_case_underscore_separated please

**Comment by TheSilentOne1** [comment]
Renamed maxPanelSize to max_panel_size.

**Pull request merged**